### ISSUE TITLE
[import/no-cycle] Doesn't detect aliased imports

### ISSUE THREAD

**Issue by christophermark** [benmosher/eslint-plugin-import#1554]
## Issue
`eslint-plugin-import` has an issue with aliased imports not being detected by the rule `import/no-cycle`.

This following example shows an import with a circular dependency that goes undetected when you change the import from a relative path, `../directory/file`, to an aliased import, `directoryAlias/file`.

Reproducing the issue
```js
/* main.js */

// import unused from '../exports/exporter'; // <-- Throws a no-cycle error when uncommented

import unused from 'exports/exporter'; // <-- Will not throw a no-cycle error, but it should!

export default function () {
  return;
}
```

To reproduce: https://github.com/christophermark/eslint-no-cycle-aliases-issue

**Comment by ljharb** [comment]
How is your alias defined/resolved?

**Comment by christophermark** [comment]
@ljharb I have the alias defined as a [local NPM package](https://github.com/christophermark/eslint-no-cycle-aliases-issue/blob/c6257541f7dbbe62f9c37b236de9cbf32f24c836/package.json#L15-L18) in my project `package.json` file:

```json
  "devDependencies": {
    "app": "file:app",
    "exports": "file:exports"
  }
```

And each directory aliased has a [very simple `package.json` file](https://github.com/christophermark/eslint-no-cycle-aliases-issue/blob/master/app/package.json) to define it as a module:

_./app/package.json_
```json
{
  "name": "app"
}
```

**Comment by ljharb** [comment]
Note that `file:` is not an *alias* - npm actually *copies* the files using `file:`, so the link might not actually end up being circular.

**Comment by christophermark** [comment]
Interesting - that seems different from what I've observed. After installing (`npm install`), the directory shows up as an alias in my `node_modules`. Is there something I'm not seeing here?

![image](https://user-images.githubusercontent.com/6303781/70443344-f43ed900-1a65-11ea-9b15-98ca2768be43.png)

**Comment by ljharb** [comment]
Hmm, it's possible that behavior's changed at some point in npm. What version of node? node has a `--preserve-symlinks` option whose default changed at some point.

**Comment by christophermark** [comment]
I'm using Node@10.15.1 and NPM@6.11.3 here

**Comment by ljharb** [comment]
Thanks! I see that if i change `app/main` to import from `../exports/exporter` instead of `exports/exporter`, it warns.

**Comment by ljharb** [comment]
I've got a fix ready, and running `eslint` on the command line verifies that. Now I'm working on adding an "invalid" test for this.

**Comment by christophermark** [comment]
Thank you @ljharb! Really appreciate the fast support and turnaround on this ❤️

**Comment by yunyu** [comment]
@ljharb Apologies for bothering you, but would you happen to have any updates on this? Would really really like to have this fixed.

**Comment by ljharb** [comment]
I'm still having trouble trying to make a test case for it that works in CI.

**Comment by ljharb** [comment]
I've filed #1588 with what I've got so far; help getting the test to fail properly is appreciated.

**Comment by jlabaj** [comment]
@ljharb could anyone help you with that PR? We have a problem in the company now after upgrading to Angular 12 the Angular CLI doesn't support detecting circular dependencies out of performance issues.
They reccomend to use linting which is not detecting the aliases as you mentioned.
https://blog.ninja-squad.com/2021/05/12/angular-cli-12.0/

**Comment by ljharb** [comment]
@jlabaj absolutely! if you can post a link to an updated branch on #1588 (not a new PR, please!) i'd be happy to pull in your changes.

**Comment by KakakuCZ** [comment]
Any update?

**Comment by jayarjo** [comment]
What happened to this effort? Still encountering this issue.

**Comment by zxybte** [comment]
You cau use `eslint-import-resolver-alias` to solve this problem. For exmaple:
import a from './src/app/goods'; => import a from '@app/goods'
// eslint-config
settings: {
    'import/resolver': {
      alias: {
        map: [['@app', './src/app']],
        extensions: ['.js', '.jsx', '.ts', '.tsx']
      },
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx']
      }
    },
  },
